internal/impl/kafka/enterprise/schema_registry_output.go

```diff
@@ -0,0 +1,195 @@
+		return
+	}
+
+	var tlsConf *tls.Config
+	var tlsEnabled bool
+	if tlsConf, tlsEnabled, err = pConf.FieldTLSToggled(sroFieldTLS); err != nil {
+		return
+	}
+
+	o.client = http.Client{}
+	if tlsEnabled && tlsConf != nil {
+		if c, ok := http.DefaultTransport.(*http.Transport); ok {
+			cloned := c.Clone()
+			cloned.TLSClientConfig = tlsConf
+			o.client.Transport = cloned
+		} else {
+			o.client.Transport = &http.Transport{
+				TLSClientConfig: tlsConf,
+			}
+		}
+	}
+
+	return
+}
+
+//------------------------------------------------------------------------------
+
+func (o *output) Connect(ctx context.Context) error {
+	if o.connected.Load() {
+		return nil
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, o.url.JoinPath("mode").String(), nil)
+	if err != nil {
+		return fmt.Errorf("failed to construct mode HTTP request: %s", err)
+	}
+	resp, err := o.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to execute mode HTTP request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("received invalid status: %d", resp.StatusCode)
+	}
+
+	var payload struct {
+		Mode string
+	}
+	if data, err := io.ReadAll(resp.Body); err != nil {
+		return fmt.Errorf("failed to read mode HTTP request body: %s", err)
+	} else {
+		if err := json.Unmarshal(data, &payload); err != nil {
+			return fmt.Errorf("failed to unmarshal response: %s", err)
+		}
+	}
+
+	if payload.Mode != "READWRITE" {
+		return fmt.Errorf("schema registry instance mode must be set to READWRITE instead of %q", payload.Mode)
+	}
+
+	o.connected.Store(true)
+
+	return nil
+}
+
+func (o *output) Write(ctx context.Context, m *service.Message) error {
+	if !o.connected.Load() {
+		return service.ErrNotConnected
+	}
+
+	var subject string
+	var err error
+	if subject, err = o.subject.TryString(m); err != nil {
+		return fmt.Errorf("failed subject interpolation: %w", err)
+	}
+
+	var b []byte
+	if b, err = m.AsBytes(); err != nil {
+		return fmt.Errorf("failed to extract message bytes: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, o.url.JoinPath("subjects", subject, "versions").String(), bytes.NewReader(b))
+	if err != nil {
+		return fmt.Errorf("failed to construct request: %s", err)
+	}
+	req.Header.Set("Content-Type", "application/vnd.schemaregistry.v1+json")
+
+	resp, err := o.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to execute request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		if respData, err := httputil.DumpResponse(resp, true); err != nil {
+			return fmt.Errorf("failed to read response: %s", err)
+		} else {
+			o.log.Debugf("Failed to push data to SchemaRegistry with status %d: %s", resp.StatusCode, string(respData))
+		}
+
+		return fmt.Errorf("request returned status: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (o *output) Close(_ context.Context) error { return nil }
```

Comment:
connected to false?

Reply:
The check in the `Connect()` method is a bit superfluous because `Write()` will error out anyway if the sink Schema Registry isn't writable so I thought it should only run once, but, then again, someone could flip SR to `READWRITE` while Connect is running, so yeah, good suggestion! Also, it's less surprising for a reviewer.